### PR TITLE
CI against rails 7.2/8.0 and PG 17, drop rails 6.1 and PG 12

### PR DESCRIPTION
**Pull request by wata727**
- Add rails 7.2 and 8.0 to the test matrix
- Add PG 17 to the test matrix
- Drop rails 6.1 support (EOL)
- Drop PG 12 support (EOL)

### CODE REVIEW

File: spec/activerecord-tenant-level-security/schema_dumper_spec.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@
     describe ".dump" do
       let(:definition) do
         io = StringIO.new
-        ActiveRecord::SchemaDumper.dump(ActiveRecord::Base.connection, io)
+        from = ActiveRecord.gem_version > Gem::Version.new("7.2.0") ? ActiveRecord::Base.connection_pool : ActiveRecord::Base.connection
+        ActiveRecord::SchemaDumper.dump(from, io)
```
Comment (wata727) [inline spec/activerecord-tenant-level-security/schema_dumper_spec.rb:7]: https://github.com/rails/rails/pull/51162

File: spec/activerecord-tenant-level-security/tenant_level_security_spec.rb [hunk 2]
```diff
@@ -154,7 +154,7 @@
 
       expect(TenantLevelSecurity.current_session_tenant_id).to eq ''
       # Returns all active connections so that checkout occurs
-      ActiveRecord::Base.clear_active_connections!
+      ActiveRecord::Base.connection_handler.clear_active_connections!
```
Comment (wata727) [inline spec/activerecord-tenant-level-security/tenant_level_security_spec.rb:157]: https://github.com/rails/rails/pull/51146

File: spec/activerecord-tenant-level-security/connection_pool_spec.rb [hunk 3]
```diff
@@ -35,8 +35,5 @@ def active_connections
         expect(Employee.all).to contain_exactly(have_attributes(name: 'Tom'))
       }.join
     end
-
-    # Ensure connections were reused
-    expect(active_connections.size).to eq 3
```
Comment (wata727) [inline spec/activerecord-tenant-level-security/connection_pool_spec.rb:40]: In Rails 8.0, `active_connections.size` is 1. After the first thread spawn, the dead threads remain as the owners, but the second time, the owners become `nil`. 

I don't know why this happens, but the following makes it clear that the connection is being reused:
https://github.com/kufu/activerecord-tenant-level-security/blob/3ddd4c89135a21d6c0e408aa9e9c5002b119fd4e/spec/activerecord-tenant-level-security/connection_pool_spec.rb#L30-L31

So simply removed the assertion.

**Review by mkmn — APPROVED** [review]
👍

**Review by sato-s — APPROVED** [review]
👍

**Pull request merged**